pkgs/applications/emulators/libretro/cores/mame2016.nix

```diff
@@ -1,5 +1,6 @@
 {
   lib,
+  gcc13Stdenv,
```

Comment:
for this one I get this error:
```
../../../../../3rdparty/rapidjson/include/rapidjson/document.h:317:82: error: assignment of read-only member 'rapidjson::GenericStringRef<CharType>::length'
  317 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                  
```
which doesn't break the build right away so have to search for it in the log.

rapidjson is a header only library so it is pretty easy to replace.

```diff
diff --git a/pkgs/applications/emulators/libretro/cores/mame2016.nix b/pkgs/applications/emulators/libretro/cores/mame2016.nix
index 31e929b7089a..5b3efdb1ea88 100644
--- a/pkgs/applications/emulators/libretro/cores/mame2016.nix
+++ b/pkgs/applications/emulators/libretro/cores/mame2016.nix
@@ -3,6 +3,7 @@
   alsa-lib,
   fetchFromGitHub,
   mkLibretroCore,
+  rapidjson,
   python3,
 }:
 mkLibretroCore {
@@ -16,6 +17,11 @@ mkLibretroCore {
     hash = "sha256-IsM7f/zlzvomVOYlinJVqZllUhDfy4NNTeTPtNmdVak=";
   };
 
+  postPatch = ''
+    rm -r 3rdparty/rapidjson
+    ln -s ${lib.getInclude rapidjson} 3rdparty/rapidjson
+  '';
+
   patches = [ ./patches/mame2016-python311.patch ];
   extraNativeBuildInputs = [ python3 ];
   extraBuildInputs = [ alsa-lib ];

```

built for me using gcc-14.  I noticed that there are a ton of things that could also get unvendored.
eg:
in the makefile I saw
```
# USE_SYSTEM_LIB_EXPAT = 1
  # USE_SYSTEM_LIB_ZLIB = 1
  # USE_SYSTEM_LIB_JPEG = 1
  # USE_SYSTEM_LIB_FLAC = 1
  # USE_SYSTEM_LIB_LUA = 1
  # USE_SYSTEM_LIB_SQLITE3 = 1
  # USE_SYSTEM_LIB_PORTMIDI = 1
  # USE_SYSTEM_LIB_PORTAUDIO = 1
  # USE_SYSTEM_LIB_UV = 1
  # USE_BUNDLED_LIB_SDL2 = 1
```
so could use nixpkgs for all of those too -- but that is out of scope for this PR.